filters/gender_bias/filter.py

```diff
@@ -0,0 +1,348 @@
+                and len(intersection_masculine) > 0
+            )
+
+            # If the processed_sentence didn't contain the keywords neither from feminine, nor from masculine arrays, set a neutral_flag value
+            neutral_flag = (
+                len(intersection_feminine) == 0
+                and len(intersection_masculine) == 0
+            )
+
+            # Use the union_flag value to set the neutral_flag value, setting to False the feminine and masculine flags
+            if union_flag is True:
+                feminine_flag = False
+                masculine_flag = False
+                neutral_flag = True
+
+            # Create the sentence object with the retrieved flag values
+            sentence_object = {
+                "sentence": sentence,
+                "feminine_flag": feminine_flag,
+                "masculine_flag": masculine_flag,
+                "neutral_flag": neutral_flag,
+            }
+
+            # Append the object to the array we return
+            flagged_sentences.append(sentence_object)
+
+        return flagged_sentences
+
+    @staticmethod
+    def count_genders(flagged_corpus):
+        """
+        count the number of sentences in each of groups
+        :param flagged_corpus: array of flagged sentences
+        :return: 3 integer values, representing feminine, masculine and neutral groups respectively
+        """
+        feminine_count = len(
+            [
+                flag
+                for flag in flagged_corpus
+                if flag.get("feminine_flag") is True
+            ]
+        )
+        masculine_count = len(
+            [
+                flag
+                for flag in flagged_corpus
+                if flag.get("masculine_flag") is True
+            ]
+        )
+        neutral_count = len(
+            [
+                flag
+                for flag in flagged_corpus
+                if flag.get("neutral_flag") is True
+            ]
+        )
+        return feminine_count, masculine_count, neutral_count
+
+    @staticmethod
+    def sort_groups(flagged_corpus):
+        """
+        sort the sentences in each of 3 groups
+        :param flagged_corpus: array of flagged sentences
+        :return: 3 arrays of strings, containing feminine, masculine and neutral groups respectively
+        """
+        feminine_group = [
+            flag.get("sentence")
+            for flag in flagged_corpus
+            if flag.get("feminine_flag") is True
+        ]
+        masculine_group = [
+            flag.get("sentence")
+            for flag in flagged_corpus
+            if flag.get("masculine_flag") is True
+        ]
+        neutral_group = [
+            flag.get("sentence")
+            for flag in flagged_corpus
+            if flag.get("neutral_flag") is True
+        ]
+
+        return feminine_group, masculine_group, neutral_group
+
+    def filter(self, sentences: []) -> bool:
+        biased = False
```

Comment:
We don't need this as we are already setting the value below. (in line 344-346)

Reply:
Fixed.